src/test/ui/rfc-2005-default-binding-mode/const.stderr

```diff
@@ -1,8 +1,16 @@
 error[E0308]: mismatched types
   --> $DIR/const.rs:14:9
    |
+LL | const FOO: Foo = Foo{bar: 5};
+   | ----------------------------- `FOO` defined here
+...
 LL |         FOO => {},
-   |         ^^^ expected `&Foo`, found struct `Foo`
+   |         ^^^
+   |         |
+   |         expected `&Foo`, found struct `Foo`
+   |         help: use different name to introduce a new binding: `other_foo`
```

Comment:
Should this also suggest `other_foo if other_foo == FOO => {}`?

Reply:
The diff seems to be gone so I don't have the context to answer, but this seems invasive to suggest a guard due to the way pattern type checking is setup, so I'd say no for now.